src/NextGen/Gateways/Stripe/NextGenStripeGateway/nextGenStripeGateway.tsx

```diff
@@ -1,20 +1,64 @@
-import {loadStripe, Stripe, StripeElements} from '@stripe/stripe-js';
+import {
+    loadStripe,
+    Stripe,
+    StripeElements,
+    StripeElementsOptionsMode,
+    StripePaymentElementChangeEvent,
+} from '@stripe/stripe-js';
 import {Elements, PaymentElement, useElements, useStripe} from '@stripe/react-stripe-js';
 import type {Gateway, GatewaySettings} from '@givewp/forms/types';
 
+let stripePromise = null;
+let stripePaymentMethod = null;
+let stripePaymentMethodIsCreditCard = false;
+
+// @see https://stripe.com/docs/currencies#zero-decimal
+const zeroDecimalCurrencies = [
+    'BIF',
+    'CLP',
+    'DJF',
+    'GNF',
+    'JPY',
+    'KMF',
+    'KRW',
+    'MGA',
+    'PYG',
+    'RWF',
+    'UGX',
+    'VND',
+    'VUV',
+    'XAF',
+    'XOF',
+    'XPF',
+];
+
+/**
+ * Takes in an amount value in dollar units and returns the calculated cents amount
+ *
+ * @unreleased
+ */
+const dollarsToCents = (amount: string, currency: string) => {
+    if (zeroDecimalCurrencies.includes(currency)) {
+        return parseInt(amount);
+    }
+
+    return parseFloat(amount) * 100;
+};
```

Comment:
@jonwaldstein Rather than maintaining a list of special currency types (there are also 3-digit currencies), I suggest using [Intl.NumberFormat.formatToParts()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts). Using this you can stripe everything away but the "integer" values and end up with any currency in cents.

Reply:
@JasonTheAdams If the currency was zero decimal wouldn't this still try to tack on the additional zeros at the end?
 In which case we would still need to check for the currency anyway to prevent that.

Reply:
Not sure I follow. Why would it tack on additional zeros? If its a zero decimal currency then this function would omit the decimals entirely and automatically round if there are any decimals.

Reply:
NumberFormatter doesn't have a concept of _zero decimal currencies_, it tries to format based on your locale & options - so if you give it an integer it would still have a fraction of `00`.

<img width="789" alt="Screenshot 2023-03-28 at 5 05 04 PM" src="https://user-images.githubusercontent.com/10138447/228366662-3c7d8e0d-9735-471e-8b0c-4ab5fe75212f.png">